### ISSUE TITLE
Chaining three tasks

### ISSUE THREAD

**Issue by kpgarrod** [firebase/firebase-queue#30]
I am trying out firebase-queue and mostly enjoying the experience so far :).

I am experiencing one issue though. When I chain tasks, say: 

>{
  "task_a" : {
    "error_state" : "queue_error_queue",
    "finished_state" : "task_b",
    "in_progress_state" : "task_a_in_progress",
    "retries" : 5,
    "start_state" : "task_a",
    "timeout" : 300000
  },
  "task_b" : {
    "error_state" : "queue_error_queue",
    "finished_state" : "task_c",
    "in_progress_state" : "task_b_in_progress",
    "start_state" : "task_b",
    "timeout" : 5000
  },
  "task_c" : {
    "in_progress_state" : "task_c_in_progress",
    "start_state" : "task_c",
    "timeout" : 5000
  }
}

When I start with `task_a`, `task_a` and `task_b` run as I expect. When I start with `task_b`,  `task_b` runs as expected, but `task_c` does not.

The task is left in this state:

>{
  "_progress" : 100,
  "_state" : "task_c",
  "_state_changed" : 1438093533931,
  "myData" : {content: 'removed'  }
} 

If I restart the server, the task runs to completion as I expect.

Any suggestions please?

**Comment by drtriumph** [comment]
I'm assuming `myData` is populated with your data - `{}` pushed to a location in Firebase causes the key to be removed.

Assuming that, it doesn't look like the task is being claimed. You can add:
```js
var winston = require('winston');
winston.level = 'debug';
```
Before you initialize the queue to get better debug logging. (you need to npm install winston)

What version of Node are you running? Some `0.11` versions are based on a V8 build with a broken sha implementation, which causes issues in the underlying Firebase implementation.

Besides that, you could also add:
```js
var Firebase = require('firebase');
Firebase.enableLogging(true);
```
Again before initializing the queue, which will turn on the underlying Firebase debug logs

**Comment by kpgarrod** [comment]
@drtriumph: thanks for the response. Yes, `myData` has valid content, I have corrected the comment above to indicate that. I am using node.js v0.12.0.

**Comment by kpgarrod** [comment]
@drtriumph: I spent a couple of hours with the debug messages without success. Eventually I created a copy of my code for the failing task, in a new file and it miraculously works! It's completely baffling for me, but I guess not something that anybody else needs to concern themselves with, so I will close the issue. Thanks again for the support.

**Comment by kpgarrod** [comment]
I looked at the logs for a couple of hours and couldn't figure out what was going on. Eventually I stumbled upon a rather strange hack. 

I currently have five Queues which are instantiated when the app runs. The failing Queue was instantiated late in the list.

If I move it to the top of the list, all five Queues work correctly!

I guess the issue must be application-related, so I would be happy to close the issue for now .
